packages/fluentui/react-northstar/src/utils/positioner/usePopper.ts

```diff
@@ -0,0 +1,408 @@
+
+            return () => {};
+          },
+          requires: [],
+        },
+
+        { name: 'flip', options: { flipVariations: true } },
+
+        /**
+         * unstable_pinned disables the flip modifier by setting flip.enabled to false; this
+         * disables automatic repositioning of the popper box; it will always be placed according to
+         * the values of `align` and `position` props, regardless of the size of the component, the
+         * reference element or the viewport.
+         */
+        unstable_pinned && { name: 'flip', enabled: false },
+
+        /**
+         * When the popper box is placed in the context of a scrollable element, we need to set
+         * preventOverflow.escapeWithReference to true and flip.boundariesElement to 'scrollParent'
+         * (default is 'viewport') so that the popper box will stick with the targetRef when we
+         * scroll targetRef out of the viewport.
+         */
+        hasScrollableElement && { name: 'flip', options: { boundary: 'clippingParents' } },
+        hasScrollableElement && { name: 'preventOverflow', options: { boundary: 'clippingParents' } },
+
+        offsetModifier,
+
+        ...(typeof userModifiers === 'function' ? userModifiers(target, container, arrow) : userModifiers),
+
+        /**
+         * This modifier is necessary to retain behaviour from popper v1 where untethered poppers are allowed by
+         * default. i.e. popper is still rendered fully in the viewport even if anchor element is no longer in the
+         * viewport.
+         */
+        unstable_disableTether && {
+          name: 'preventOverflow',
+          options: { altAxis: unstable_disableTether === 'all', tether: false },
+        },
+
+        flipBoundary && {
+          name: 'flip',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, flipBoundary),
+          },
+        },
+        overflowBoundary && {
+          name: 'preventOverflow',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+        },
+
+        {
+          name: 'onUpdate',
+          enabled: true,
+          phase: 'afterWrite' as PopperJs.ModifierPhases,
+          fn: handleStateUpdate,
+        },
+
+        autoSize && {
+          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
+          name: 'applyMaxSize',
+          enabled: true,
+          phase: 'beforeWrite' as PopperJs.ModifierPhases,
+          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+          fn({ state, options }: PopperJs.ModifierArguments<{}>) {
```

Comment:
There are different scopes of `options` in this hook
```suggestion
          fn({ state, modifierOptions }: PopperJs.ModifierArguments<{}>) {
```

Reply:
This change is not valid as this callback will be called by Popper.

Reply:
you can change the name through destructuring: `{state, options: modifierOptions}`

It's what I've had to do porting this to converged

Reply:
Got it, applied